syntax = "proto3";

option java_package = "edu.wisc.cs.unreliable_afs";

package unreliable_afs;

service UnreliableAFSProto {
    // rpc GetAttr (GetAttrReq) returns (GetAttrReply) {}
    rpc Mkdir (MkdirRequest) returns (MkdirReply) {}
    rpc Rmdir (RmdirRequest) returns (RmdirReply) {}
}

message MkdirRequest {
    string path = 1;
    int32 mode = 2;
}

message MkdirReply {
    int32 err = 1;
}

message RmdirRequest {
    string path = 1;
}

message RmdirReply {
    int32 err = 1;
}